lib/phoenix/endpoint/instrument.ex

```diff
@@ -72,11 +74,16 @@ defmodule Phoenix.Endpoint.Instrument do
       end
 
       # Catch-all clause
-      def instrument(_event, compile, _runtime, fun)
-          when is_map(compile) and is_function(fun, 0) do
+      def instrument(_event, compile, runtime, fun)
+          when is_map(compile) and is_map(runtime) and is_function(fun, 0) do
         fun.()
       end
 
+      # Bad args clauses
+      def instrument(_event, _compile, runtime, _fun) do
```

Comment:
That's not necessarily true, it can be invoked because compile for some reasons is not a map, fun is not a function, etc. Just yank this clause. :)